airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -38,41 +39,32 @@ def connector_spec_dict_fixture(actual_connector_spec):
     return json.loads(actual_connector_spec.json())
 
 
-@pytest.fixture(name="actual_connector_spec")
-def actual_connector_spec_fixture(request: BaseTest, docker_runner: ConnectorRunner) -> ConnectorSpecification:
-    if not request.instance.spec_cache:
-        output = docker_runner.call_spec()
-        spec_messages = filter_output(output, Type.SPEC)
-        assert len(spec_messages) == 1, "Spec message should be emitted exactly once"
-        spec = spec_messages[0].spec
-        request.spec_cache = spec
-    return request.spec_cache
-
-
-@pytest.fixture(name="previous_connector_spec")
-def previous_connector_spec_fixture(
-    request: BaseTest, previous_connector_docker_runner: ConnectorRunner
-) -> Optional[ConnectorSpecification]:
-    if previous_connector_docker_runner is None:
-        logging.warning(
-            "\n We could not retrieve the previous connector spec as a connector runner for the previous connector version could not be instantiated."
-        )
-        return None
-    if not request.instance.previous_spec_cache:
-        output = previous_connector_docker_runner.call_spec()
-        spec_messages = filter_output(output, Type.SPEC)
-        assert len(spec_messages) == 1, "Spec message should be emitted exactly once"
-        spec = spec_messages[0].spec
-        request.instance.previous_spec_cache = spec
-    return request.instance.previous_spec_cache
-
-
 @pytest.mark.default_timeout(10)
 class TestSpec(BaseTest):
 
     spec_cache: ConnectorSpecification = None
     previous_spec_cache: ConnectorSpecification = None
 
+    @staticmethod
+    def compute_spec_diff(actual_connector_spec: ConnectorSpecification, previous_connector_spec: ConnectorSpecification):
+        return DeepDiff(previous_connector_spec.dict(), actual_connector_spec.dict(), view="tree")
+
+    @pytest.fixture(name="spec_diff")
+    def spec_diff_fixture(
+        self, inputs: SpecTestConfig, actual_connector_spec: ConnectorSpecification, previous_connector_spec: ConnectorSpecification
+    ) -> DeepDiff:
+        if (
+            inputs.backward_compatibility_tests_config.disable_backward_compatibility_tests_for_version
+            == inputs.backward_compatibility_tests_config.previous_connector_version
+        ):
+            pytest.skip(
+                f"Backward compatibility tests are disabled for version {inputs.backward_compatibility_tests_config.disable_backward_compatibility_tests_for_version}."
+            )
+        if previous_connector_spec is None:
+            pytest.skip("The previous connector spec could not be retrieved.")
```

Comment:
👍🏻 any tests using this `spec_diff` fixture should get skipped because of this

Reply:
I moved the skipping logic to a separate fixture for the separation of concerns.